internal/safety/safety_bench_test.go

```diff
@@ -0,0 +1,140 @@
+package safety
+
+import (
+	"testing"
+
+	"github.com/vdaas/vald/internal/errors"
+)
+
+func BenchmarkRecoverFunc(b *testing.B) {
+	type args struct {
+		fn func() error
+	}
+	type test struct {
+		name string
+		args args
+	}
+	tests := []test{
+		{
+			name: "fn return error",
+			args: args{
+				fn: func() error {
+					return errors.New("fn err")
+				},
+			},
+		},
+	}
+	for _, test := range tests {
+		b.Run(test.name, func(b *testing.B) {
+			b.RunParallel(func(pb *testing.PB) {
+				for pb.Next() {
+					got := RecoverFunc(test.args.fn)
+					if got == nil {
+						b.Error("got is empty")
+					}
+				}
+			})
+		})
+	}
+}
+
+func BenchmarkRecoverWithoutPanicFunc(b *testing.B) {
+	type args struct {
+		fn func() error
+	}
+	type test struct {
+		name string
+		args args
+	}
+	tests := []test{
+		{
+			name: "fn return error",
+			args: args{
+				fn: func() error {
+					return errors.New("fn err")
+				},
+			},
+		},
+	}
+	for _, test := range tests {
+		b.Run(test.name, func(b *testing.B) {
+			b.RunParallel(func(pb *testing.PB) {
+				for pb.Next() {
+					got := RecoverWithoutPanicFunc(test.args.fn)
+					if got == nil {
+						b.Error("got is empty")
+					}
+				}
+			})
+		})
+	}
+}
+
+func BenchmarkRecoverWithoutPanicFunc_with_execution(b *testing.B) {
```

Comment:
is the benchmark name here ok?

Reply:
what's the meaning of `with execution`?

Reply:
we do not create such as agreement 🤔 
you should  name just `BenchmarkRecoverWithoutPanicFunc`
@hlts2 how do you think?

Reply:
> you should name just BenchmarkRecoverWithoutPanicFunc
@hlts2 how do you think?

I think it looks good to me 👍

Reply:
It conflicts with the benchmark function on line 41.

Why I named like this is because the `RecoverWithoutPanicFunc` return the function with panic handler, and this function benchmark the execution of returning function.
It is different with `BenchmarkRecoverWithoutPanicFunc` which only benchmark the `RecoverWithoutPanicFunc` function

Reply:
Got It!!
btw `BenchmarkRecoverWithoutPanicFunc` is a benchmark test that returns a function, and even if we get it, we can't get meaningful data, so how about putting it together?

Reply:
for example, It's like rewriting the contents of `BenchmarkRecoverWithoutPanicFunc` to the contents of `BenchmarkRecoverWithoutPanicFunc_with_execution`.
what do you think?

Reply:
Sorry but I am not sure what do you mean by ` It's like rewriting the contents of BenchmarkRecoverWithoutPanicFunc to the contents of BenchmarkRecoverWithoutPanicFunc_with_execution.`